test/test-misc.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# simple tests that don't deserve their own testfile
+
+# library of utility functions
+# shellcheck disable=SC1091
+. test/util.sh
+
+echo running "$0"
+
+ROOT=$(dirname "${BASH_SOURCE}")/..
+cd "${ROOT}" || exit 1
```

Comment:
if you're going to add this `|| exit 1` on here, please do it to the other `cd "${ROOT}"` 's too.

Reply:
That will be part of the Shellcheck commit which is not included in this PR. For new/changed lines I already made them shellcheck compatible, for unchanged lines I left hem in that commit for separation.